hledger-lib/Hledger/Data/MarketPrice.hs

```diff
@@ -12,22 +12,35 @@ are thousands separated by comma, significant decimal places and so on.
 
 -}
 
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings, LambdaCase #-}
 
 module Hledger.Data.MarketPrice
 where
 import qualified Data.Text as T
--- import Test.HUnit
+import Test.HUnit
 
 import Hledger.Data.Amount
-import Hledger.Data.Types
 import Hledger.Data.Dates
+import Hledger.Data.Types
+-- import Hledger.Read.JournalReader
 import Hledger.Utils
+import Hledger.Utils.Parse
 
+-- | Get the string representation of an market price, based on its
+-- commodity's display settings.
 showMarketPrice :: MarketPrice -> String
 showMarketPrice mp = unwords
     [ "P"
     , showDate (mpdate mp)
     , T.unpack (mpcommodity mp)
     , (showAmount . setAmountPrecision maxprecision) (mpamount mp)
```

Comment:
I see the issue now.  Not sure how to best resolve this.  Would this be a bug in `showAmount`, which truncates prices when it shouldn't be?